pkg/log/context.go

```diff
@@ -23,46 +23,66 @@ import (
 	"os"
 	"sync"
 
+	"github.com/onsi/ginkgo"
+
 	"github.com/onrik/logrus/filename"
+
 	"github.com/sirupsen/logrus"
 )
 
 type logKey struct{}
 
+const (
+	// FieldComponentName is the key of the component field in the log message.
+	FieldComponentName = "component"
+	// FieldCorrelationID is the key of the correlation id field in the log message.
+	FieldCorrelationID = "correlation_id"
+)
+
 var (
+	defaultEntry = logrus.NewEntry(logrus.StandardLogger())
+
 	supportedFormatters = map[string]logrus.Formatter{
 		"json": &logrus.JSONFormatter{},
 		"text": &logrus.TextFormatter{},
 	}
-	regMutex     = sync.Mutex{}
-	once         = sync.Once{}
-	defaultEntry = logrus.NewEntry(logrus.StandardLogger())
+
+	supportedOutputs = map[string]io.Writer{
+		os.Stdout.Name(): os.Stdout,
+		os.Stdin.Name():  os.Stdin,
+		os.Stderr.Name(): os.Stderr,
+		"ginkgowriter":   ginkgo.GinkgoWriter,
```

Comment:
I'd rather not have test dependencies in productive code and leave the configuration as is.

Reply:
we need to change the output for thr ginkgo tests so that it does not spam logs for successfull tests - the old way doesnt work anymore, so we needed a way to switch the writer. I am open to suggestions